hw/ip/otbn/dv/otbnsim/sim/loop.py

```diff
@@ -47,6 +47,18 @@ def __init__(self, start_addr: int, insn_count: int, restarts: int):
         self.start_addr = start_addr
         self.match_addr = start_addr + 4 * insn_count
 
+    @property
+    def loop_insn_addr(self) -> int:
+        ''' The address of the LOOP or LOOPI instruction. '''
+        assert self.start_addr >= 4
+        return self.start_addr - 4
+
+    @property
+    def last_insn_addr_in_loop(self) -> int:
+        ''' The address of the last instruction in a loop body. '''
+        assert self.match_addr >= 4
+        return self.match_addr - 4
```

Comment:
Since you're changing the `step` function to compare `pc` with the last instruction in the loop (an improvement, thanks!), maybe it would be better to store "`last_addr`" rather than `match_addr`, which would get rid of this computation entirely?

Reply:
Yes, that's clearly better. Changed to that. (Went around in circles there a couple times.)